src/components/structures/auth/Registration.tsx

```diff
@@ -139,8 +139,20 @@ export default class Registration extends React.Component<IProps, IState> {
 
     componentDidMount() {
         this.replaceClient(this.props.serverConfig);
+        //triggers a confirmation dialog for data loss before page unloads/refreshes
+        window.addEventListener("beforeunload", this.unloadCallback);
     }
 
+    componentWillUnmount() {
+        window.removeEventListener("beforeunload", this.unloadCallback);
+    }
+
+    private unloadCallback = (event) => {
```

Comment:
`event` should have a type annotation

Reply:
I couldn't think of a proper event type for this, what do you think it should be?

Reply:
Done